src/hooks/useWorkdayData.tsx

```diff
@@ -45,17 +52,20 @@ export default function useWorkdayData(
               };
             });
 
-      const counts = workdays?.reduce(
-        (counts, { demographics }) => {
-          return (
-            demographics?.reduce(
-              (counts, { type, amount }) => ({ ...counts, [type]: counts[type] + amount }),
-              counts
-            ) ?? counts
-          );
-        },
-        { gender: 0, age: 0, ethnicity: 0 }
-      );
+      const initialCounts: DemographicCounts = {
+        gender: 0,
+        age: 0,
+        ethnicity: 0,
+        caste: 0
```

Comment:
We want to avoid sending `caste` at all for non-HBF frameworks, and avoid sending `ethnicity` for HBF frameworks, so this needs to be framework-aware as well.